### PR TITLE
feat: add bc-pay-button web component

### PR DESCRIPTION
**Pull request by rolznz**
Fixes https://github.com/getAlby/bitcoin-connect/issues/155

- [x] change button state after payment completed
- [x] update react wrapper

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -187,6 +187,15 @@ _Use another popular framework? please let us know or feel free to create a PR f
 Bitcoin Connect exposes the following web components for allowing users to connect their desired Lightning wallet:
 
 - `<bc-button/>` - launches the Bitcoin Connect Modal on click
+  - Arguments:
+    - `title` - (optional) change the title of the button
+- `<bc-pay-button/>` - launches the Bitcoin Connect Payment Modal on click
+  - Arguments:
+    - `invoice` - BOLT11 invoice
```
Comment (bumi) [inline README.md:194]: could this be a function to dynamically load the invoice on click? 
and avoid the requirement to have an invoice on page load.

Comment (rolznz) [reply README.md]: fixed

File: README.md [hunk 2]
```diff
@@ -93,6 +93,14 @@ init({
   const {preimage} = await provider.sendPayment("lnbc...");
 }}/>
 
+// render a "Pay Now" button
+// invoice can be unset initially - using the onClick function is a good time to fetch the invoice
+// set the `payment` prop to override the payment status if a payment was made externally
+<Button invoice={invoice} onClick={() => {
```
Comment (bumi) [inline README.md:99]: is this PayButton?

File: src/components/bc-pay-button.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      <bci-button variant="primary">
+        ${isLoading
+          ? html`<span class="ml-1 mr-1">${loadingIcon}</span>`
+          : this._paid
+          ? html`<span class="-ml-0.5">${checkIcon}</span>`
+          : html`<span class="-ml-0.5">${bcIcon}</span>`}
+        <span class="font-semibold">
+          ${isLoading
+            ? html`Loading...`
+            : html`${this._paid ? 'Paid' : this.title}`}
```
Comment (bumi) [inline src/components/bc-pay-button.ts:65]: can children be a way to customize the text?

defaults like "Paid" will likely not work generally

File: src/components/bc-pay-button.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  override title = 'Pay Now';
+
+  @property()
+  invoice?: string;
+
+  @property({})
+  preimage?: string;
+
+  @state()
+  _waitingForInvoice = false;
```
Comment (bumi) [inline src/components/bc-pay-button.ts:27]: what is waiting for invoice?

File: react/src/hooks/useOnPaid.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  payment?: SendPaymentResponse
+) {
+  React.useEffect(() => {
+    const onPaidEventHandler = (event: Event) => {
+      onPaid?.((event as CustomEvent).detail);
+    };
+    window.addEventListener('bc:onpaid', onPaidEventHandler);
+
+    if (payment) {
+      // TODO: remove once bc-send-payment accepts preimage
```
Comment (bumi) [inline react/src/hooks/useOnPaid.ts:15]: it does in this PR, doesn't it?

File: src/components/bc-pay-button.ts (hunk 4 above)
Comment (rolznz) [reply src/components/bc-pay-button.ts:27]: if the button is clicked when the invoice attribute is not set, like how it works on https://bitcoin-connect.com now. I'll add a comment.

File: react/src/hooks/useOnPaid.ts (hunk 5 above)
Comment (rolznz) [reply react/src/hooks/useOnPaid.ts:15]: https://github.com/getAlby/bitcoin-connect/issues/162

File: README.md (hunk 2 above)
Comment (rolznz) [reply README.md]: thanks, fixed

**Pull request merged**